src/Platform/Microsoft.Testing.Platform/OutputDevice/Console/TestModule.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Testing.Platform.OutputDevice.Console;
+
+internal sealed class TestModule
+{
+    public TestModule(int slotIndex, string assembly, string? targetFramework, string? architecture, StopwatchAbstraction stopwatch)
+    {
+        SlotIndex = slotIndex;
+        Assembly = assembly;
+        TargetFramework = targetFramework;
+        Architecture = architecture;
+        Stopwatch = stopwatch;
+    }
+
+    public int SlotIndex { get; }
+
+    public string Assembly { get; }
+
+    public string? TargetFramework { get; }
+
+    public string? Architecture { get; }
+
+    public StopwatchAbstraction Stopwatch { get; }
+
+    public List<string> Attachments { get; } = new();
+
+    public List<IProgressMessage> Messages { get; } = new();
```

Comment:
Could we expose as "readonly"/"immutable" to avoid having anything that can mutate this?

Reply:
It can be done, but it feels like too much ceremony for this simple internal object that just holds the state and is not passed outside of the logger.